packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol

```diff
@@ -269,66 +267,143 @@ contract LongShortPair is Testable, Lockable {
      *        GLOBAL STATE FUNCTIONS        *
      ****************************************/
 
-    function expire() public postExpiration() onlyOpenState() nonReentrant() {
-        _requestOraclePriceExpiration();
-        contractState = ContractState.ExpiredPriceRequested;
+    /**
+     * @notice Enables the LPS to request early expiration. This initiates a price request to the optimistic oracle at
+     * the provided timestamp with a modified version of the ancillary data that includes the key "earlyExpiration:1"
+     * which signals to the OO that this is an early expiration request, rather than standard settlement.
+     * @dev Will revert if: a) the contract is already early expire, b) it is after the expiration timestamp, c)
+     * early expiration is disabled for this contract, d) the proposed expiration timestamp is in the future.
+     * e) an early expiration attempt has already been made (in pending state).
+     * @param _earlyExpirationTimestamp timestamp at which the early expiration is proposed.
+     */
+    function requestEarlyExpiration(uint64 _earlyExpirationTimestamp)
```

Comment:
hopefully this is simple enough. I think it's about as clean as we'll get it.

note that this does _not_ enable the requestor to propose a price at the same time. this was done intentionally to mimic the same flow with _all_ other settlement within uma contracts (i.e first settle is called and separately a price is proposed)